charts/shoot-addons/charts/kube2iam/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: {{ include "networkpolicyversion" . }}
+kind: NetworkPolicy
+metadata:
+  name: {{ template "fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    addonmanager.kubernetes.io/mode: Reconcile
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "name" . }}
+      release: {{ .Release.Name }}
+  policyTypes:
+  - Egress
+  egress:
+  - to:
+    - ipBlock:
+        cidr: 169.254.169.254/32
```

Comment:
This will allow only metadata service but block all other egress including access to "apiserver" . Did you check if it is working fine?

Reply:
The traffic to other dests is not blocked by the ``GlobalNetworkPolicy`` or any other ``NetworkPolicy``, right? This policy explicitly allows only traffic to ``169.254.169.254/32`` and overrides the rules of ``GlobalNetworkPolicy`` for the labeled app. Seems ok for me. 
@mvladev Am I right?

Reply:
NetworkPolicy applies whitelist mechanism. So, all cidr out of mentioned ranges are blocked. That's why we had to apply some workaround at https://github.com/gardener/gardener/blob/master/charts/seed-controlplane/charts/kube-apiserver/templates/network-policy.yaml#L18